presto-geospatial/src/main/java/com/facebook/presto/geospatial/BingTileFunctions.java

```diff
@@ -0,0 +1,323 @@
+        int tileY = (int) clip(y * mapSize + 0.5, 0, mapSize - 1);
+        return BingTile.fromCoordinates(tileX / TILE_PIXELS, tileY / TILE_PIXELS, zoomLevel);
+    }
+
+    private static OGCGeometry tileToPolygon(BingTile tile)
+    {
+        Point upperLeftCorner = tileXYToLatitudeLongitude(tile.getX(), tile.getY(), tile.getZoomLevel());
+        Point lowerRightCorner = tileXYToLatitudeLongitude(tile.getX() + 1, tile.getY() + 1, tile.getZoomLevel());
+
+        Polyline boundary = new Polyline();
+        boundary.startPath(upperLeftCorner);
+        boundary.lineTo(lowerRightCorner.getX(), upperLeftCorner.getY());
+        boundary.lineTo(lowerRightCorner);
+        boundary.lineTo(upperLeftCorner.getX(), lowerRightCorner.getY());
+
+        Polygon polygon = new Polygon();
+        polygon.add(boundary, false);
+
+        return OGCGeometry.createFromEsriGeometry(polygon, null);
+    }
+
+    /**
+     * @return true if the geometry is a point or a rectangle
+     */
+    private static boolean isPointOrRectangle(OGCGeometry geometry, Envelope envelope)
+    {
+        if (geometry instanceof OGCPoint) {
+            return true;
+        }
+
+        if (!(geometry instanceof OGCPolygon)) {
+            return false;
+        }
+
+        OGCPolygon polygon = (OGCPolygon) geometry;
+        if (polygon.numInteriorRing() > 0) {
+            return false;
+        }
+
+        MultiVertexGeometry multiVertexGeometry = (MultiVertexGeometry) polygon.getEsriGeometry();
+        if (multiVertexGeometry.getPointCount() != 4) {
+            return false;
+        }
+
+        Set<Point> corners = new HashSet<>();
+        corners.add(new Point(envelope.getXMin(), envelope.getYMin()));
+        corners.add(new Point(envelope.getXMin(), envelope.getYMax()));
+        corners.add(new Point(envelope.getXMax(), envelope.getYMin()));
+        corners.add(new Point(envelope.getXMax(), envelope.getYMax()));
+
+        for (int i = 0; i < 4; i++) {
+            Point point = multiVertexGeometry.getPoint(i);
+            if (!corners.contains(point)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private static void checkZoomLevel(long zoomLevel)
+    {
+        checkCondition(zoomLevel > 0, ZOOM_LEVEL_TOO_SMALL);
+        checkCondition(zoomLevel <= MAX_ZOOM_LEVEL, ZOOM_LEVEL_TOO_LARGE);
+    }
+
+    private static void checkCoordinate(long coordinate, long zoomLevel)
+    {
+        checkCondition(coordinate >= 0 && coordinate < (1 << zoomLevel), "XY coordinates for a Bing tile at zoom level %s must be within [0, %s) range", zoomLevel, 1 << zoomLevel);
+    }
+
+    private static void checkQuadKey(@SqlType(StandardTypes.VARCHAR) Slice quadkey)
+    {
+        checkCondition(quadkey.length() > 0, QUAD_KEY_EMPTY);
+        checkCondition(quadkey.length() <= MAX_ZOOM_LEVEL, QUAD_KEY_TOO_LONG);
+    }
+
+    private static void checkLatitude(double latitude, String errorMessage)
```

Comment:
you can also inline the one line methods here to where you are calling them.

Reply:
Is there any particular reason to inline these functions? They are used in multiple places and the checks are somewhat verbose. I prefer to use helper functions to avoid copy-pasting long  lines of code.

For example, after inlining

        checkLatitude(envelope.getYMin(), LATITUDE_SPAN_OUT_OF_RANGE);
        checkLatitude(envelope.getYMax(), LATITUDE_SPAN_OUT_OF_RANGE);

becomes

       checkCondition(envelope.getYMin() >= MIN_LATITUDE, envelope.getYMin() <= MAX_LATITUDE, LATITUDE_SPAN_OUT_OF_RANGE)
       checkCondition(envelope.getYMax() >= MIN_LATITUDE, envelope.getYMax() <= MAX_LATITUDE, LATITUDE_SPAN_OUT_OF_RANGE)

The latter seems harder to read.